src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -139,201 +139,430 @@ namespace OpenMS
-    ParamXMLFile paramFile;
-    paramFile.load(file, param);
-
-    UInt an(0);
-    String name, symbol;
-    Map<UInt, double> Z_to_abundancy;
-    Map<UInt, double> Z_to_mass;
-    
-    // determine prefix
-    vector<String> split;
-    param.begin().getName().split(':', split);
-    String prefix("");
-    for (Size i = 0; i < split.size() - 1; ++i)
-    {
-      prefix += split[i] + ":";
-    }
-    //cout << "first element prefix=" << prefix << endl;
-
-
-    for (Param::ParamIterator it = param.begin(); it != param.end(); ++it)
-    {
-      it.getName().split(':', split);
-      
-      // new element started?
-      if (!it.getName().hasPrefix(prefix))
-      {
-        // update prefix
-        prefix = "";
-        for (Size i = 0; i < split.size() - 1; ++i)
-        {
-          prefix += split[i] + ":";
-        }
-        // cout << "new element prefix=" << prefix << endl;
-        
-        storeElement_(an, name, symbol, Z_to_abundancy, Z_to_mass);
-
-        Z_to_abundancy.clear();
-        Z_to_mass.clear();
-      } // new element started
-
-      // top level: read the contents of the element section
-      const String& key = split[2];
-      String value = it->value;
-      value.trim();
-
-      // cout << "Key=" << key << endl;
-
-      if (key == "AtomicNumber")
-      {
-        an = (UInt)value.toInt();
-      }
-      else
-      {
-        if (key == "Isotopes")
-        {
-          UInt Z = UInt(split[3].toInt());
-          String item = split[4];
-          if (item == "RelativeAbundance")
-          {
-            Z_to_abundancy[Z] = double(value.toDouble() / 100.0);
-          }
-          else if (item == "AtomicMass")
-          {
-            Z_to_mass[Z] = double(value.toDouble());
-          }
-          else
-          {
-            cerr << "read unknown item in Isotopes: " << item << endl;
-          }
-        }
-        else
-        {
-          if (key == "Name")
-          {
-            name = value;
-          }
-          else
-          {
-            if (key == "Symbol")
-            {
-              symbol = value;
-            }
-            else
-            {
-              cerr << "read unknown tag: " << key << endl;
-            }
-          }
-        }
-      }
-    }
-
-    // build last element
-    storeElement_(an, name, symbol, Z_to_abundancy, Z_to_mass);
 
+    } 
   }
 
-  IsotopeDistribution ElementDB::parseIsotopeDistribution_(const Map<UInt, double>& Z_to_abundance, const Map<UInt, double>& Z_to_mass)
+  IsotopeDistribution ElementDB::parseIsotopeDistribution_(const map<unsigned int, double>& abundance, const map<unsigned int, double>& mass)
   {
     IsotopeDistribution::ContainerType dist;
     
-    vector<UInt> keys;
-    for (Map<UInt, double>::const_iterator it = Z_to_abundance.begin(); it != Z_to_abundance.end(); ++it)
+    vector<unsigned int> keys;
+    for (map<unsigned int, double>::const_iterator it = abundance.begin(); it != abundance.end(); ++it)
```

Comment:
with C17 this can now be written shorter as:
```
for (const auto& [key, _] : abundance) { keys.push_back(key) };
```

Reply:
could trigger an unused variable warning though so maybe the old way is better

Reply:
Okay, then I will keep it the old way.